android/java/org/chromium/chrome/browser/crypto_wallet/activities/BraveWalletDAppsActivity.java

```diff
@@ -88,6 +103,40 @@ public void finishNativeInitialization() {
         if (mActivityType == ActivityType.ADD_ETHEREUM_CHAIN) {
             addEthereumChain();
             finish();
+        } else if (mActivityType == ActivityType.CONFIRM_TRANSACTION) {
+            transactionInfos = new ArrayList<>();
+            TransactionInfo transactionInfo = new TransactionInfo();
+            transactionInfo.id = "";
+            transactionInfo.txType = TransactionType.MAX_VALUE;
+            transactionInfo.txParams = new String[] {};
+            transactionInfo.txArgs = new String[] {};
+            transactionInfo.txDataUnion = new TxDataUnion();
+
+            TxData1559 txData1559 = new TxData1559();
+            txData1559.maxPriorityFeePerGas = "0x0";
+            txData1559.baseData = new TxData();
+            txData1559.maxFeePerGas = "0x0";
+            txData1559.baseData.gasPrice = "0x0";
+            txData1559.baseData.gasLimit = "0x0";
+            txData1559.gasEstimation = new GasEstimation1559();
+            txData1559.gasEstimation.slowMaxFeePerGas = "0x0";
+            txData1559.gasEstimation.fastMaxFeePerGas = "0x0";
+            txData1559.gasEstimation.baseFeePerGas = "0x0";
+
+            txData1559.baseData.data = new byte[] {};
+            txData1559.baseData.to = "0x0";
+            txData1559.baseData.value = "0x0";
+
+            transactionInfo.txDataUnion.setEthTxData1559(txData1559);
+            transactionInfo.fromAddress = "0x0";
+
+            transactionInfos.add(transactionInfo);
+            transactionInfos.add(transactionInfo);
```

Comment:
why do we add the same transaction twice?

Reply:
That's for testing multiple transaction's approval/rejection. The approval dialog should be able to handle multiple transactions and since it's dummy data so added the same twice.